sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3730,7 +3728,9 @@ object GpuOverrides extends Logging {
       // This needs to match what murmur3 supports.
       PartChecks(RepeatingParamCheck("hash_key",
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
-            TypeSig.STRUCT).nested(), TypeSig.all)),
+            TypeSig.STRUCT + TypeSig.ARRAY).nested(),
```

Comment:
We don't support it. Which is why I added code to disallow HashAggregateExec. Is there any other way I can restrict Array of struct but allow struct of array?

Reply:
This code isn't for HashAggregateExec but ShuffleExec.  It's the hash partitioner for shuffle.

The tricky part is we want to allow structs with arrays but not arrays with structs, so. I think this will need an explicit override of `tagPartForGpu` that checks for this, along with the same PS note we added for hash aggregate.  And there should be a test in repart_test that we fallback properly to know we got the check correct.